### PR TITLE
Cadc 9824 Fixes a bug to match a subject's group memberships to the RelyParty access group.

### PR DESCRIPTION
**Pull request by yeunga**
Changed the matching algorithm to take advantage of a subject already containing group membership information and determine if the access group is in the group memberships.

### CODE REVIEW

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java [hunk 1]
```diff
@@ -266,14 +268,27 @@ public static RelyParty getRelyParty(String clientID) {
...
+    public static boolean accessAllowed(RelyParty rp, Subject subject) {
         GroupURI accessGroup = rp.getAccessGroup();
         if (accessGroup == null) {
             // access group not specified, allow access
             return true;
         } else {
-            GroupClient groupClient = GroupUtil.getGroupClient(accessGroup.getServiceID());
-            return groupClient.isMember(accessGroup);
+            subject = AuthenticationUtil.augmentSubject(subject);
+            Set<GroupMemberships> groupMembershipsSet = subject.getPrivateCredentials(GroupMemberships.class);
```
Comment (brianmajor) [inline cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java:278]: The use of private credentials in the subject to store memberships is a implementation detail/secret in the LDAP persistence layer and should not be used externally like this.  Instead, construct an LdapGroupPersistence object and call `getGroups(Role.MEMBER, GroupURI.toString())`. If the single group comes back in the collection then they are a member.  If not they are not a member.

**Review by brianmajor — CHANGES_REQUESTED** [review]

File: cadc-access-control-server/ac-oidc-clients.properties [hunk 2]
```diff
@@ -1 +1,11 @@
-arbutus-harbor=harbor-secret
+arbutus-harbor.secret = harbor-secret
```
[the remark this answers is not in the document]
Comment (yeunga) [reply cadc-access-control-server/ac-oidc-clients.properties:1]: Okay. It's better that I just delete this file.

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java [hunk 3]
```diff
@@ -266,14 +268,27 @@ public static RelyParty getRelyParty(String clientID) {
...
+            for (GroupMemberships groupMemberships : groupMembershipsSet) {
+                List<Group> groups = groupMemberships.getMemberships(Role.MEMBER);
+                for (Group group : groups) {
+                    GroupURI groupURI = group.getID();
+                    log.info("alinga-- group: " + groupURI);
```
[the remark this answers is not in the document]
Comment (yeunga) [reply cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java]: Done.

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java (hunk 1 above)
Comment (yeunga) [reply cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java]: Reworked.

**Review by brianmajor — APPROVED** [review]

**Pull request merged**